### PR TITLE
Add ugc datatypes.

### PR DESCRIPTION
**Pull request by mgsergio**
@ygorshenin

### CODE REVIEW

File: ugc/api.hpp [hunk 1]
```diff
@@ -12,17 +14,17 @@ namespace ugc
 class Api
 {
 public:
-  // TODO (@y, @mgsergio): replace void() by void(UGC const &).
-  using Callback = std::function<void()>;
+  using StaticUGCCallback = std::function<void(UGC const &)>;
+  using DynamicUGCCallback = std::function<void(UGCUpdate const &)>;
```
Comment (ygorshenin) [inline ugc/api.hpp]: UGCUpdateCallback

File: ugc/ugc_types.hpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#include "indexer/feature_decl.hpp"
+
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <map>
+#include <vector>
+
+
```
Comment (ygorshenin) [inline ugc/ugc_types.hpp]: No need in this blank line.

File: ugc/ugc_types.hpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+namespace ugc
+{
+struct RatingRecord
+{
+  std::string m_key;
+  float m_value;
```
Comment (ygorshenin) [inline ugc/ugc_types.hpp]: Please, default-initialize these values.

File: ugc/CMakeLists.txt [hunk 4]
```diff
@@ -4,6 +4,7 @@ set(
   SRC
   api.cpp
   api.hpp
+  ugc_types.hpp
```
Comment (ygorshenin) [inline ugc/CMakeLists.txt]: Please, get rid of this 'ugc_' prefix.

File: ugc/api.hpp [hunk 5]
```diff
@@ -12,17 +14,17 @@ namespace ugc
 class Api
 {
 public:
-  // TODO (@y, @mgsergio): replace void() by void(UGC const &).
-  using Callback = std::function<void()>;
+  using StaticUGCCallback = std::function<void(UGC const &)>;
```
Comment (ygorshenin) [inline ugc/api.hpp]: UGCCallback

File: ugc/api.hpp [hunk 6]
```diff
@@ -12,17 +14,17 @@ namespace ugc
...
+  using DynamicUGCCallback = std::function<void(UGCUpdate const &)>;
 
   explicit Api(Index const & index);
 
-  void GetStaticUGC(FeatureID const & id, Callback callback);
-  void GetDynamicUGC(FeatureID const & id, Callback callback);
+  void GetStaticUGC(FeatureID const & id, StaticUGCCallback callback);
```
Comment (ygorshenin) [inline ugc/api.hpp]: Тогда просто GetUGC и GetUGCUpdate. То же самое для Impl.

File: ugc/ugc_types.hpp [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+struct Rating
+{
+  std::vector<RatingRecord> m_ratings;
+  float m_aggValue;
+};
+
+class UID
+{
+  std::string toString() const;
```
Comment (ygorshenin) [inline ugc/ugc_types.hpp]: This is private too. Please, rename to "ToString".

File: ugc/CMakeLists.txt (hunk 4 above)
Comment (mgsergio) [inline ugc/CMakeLists.txt]: done.

File: ugc/ugc_types.hpp (hunk 7 above)
Comment (mgsergio) [inline ugc/ugc_types.hpp]: Done.

File: ugc/ugc_types.hpp (hunk 2 above)
Comment (mgsergio) [inline ugc/ugc_types.hpp]: Done.

File: ugc/api.hpp (hunk 5 above)
Comment (mgsergio) [inline ugc/api.hpp]: Done.

File: ugc/api.hpp (hunk 1 above)
Comment (mgsergio) [inline ugc/api.hpp]: Done.

File: ugc/api.hpp (hunk 6 above)
Comment (mgsergio) [inline ugc/api.hpp]: Done.

File: ugc/ugc_types.hpp (hunk 3 above)
Comment (mgsergio) [inline ugc/ugc_types.hpp]: Done using constructors.

**Comment by mgsergio** [comment]
PTAL

**Pull request merged**